apollo-router/src/plugin/serde.rs

```diff
@@ -209,3 +210,27 @@ where
     }
     deserializer.deserialize_str(RegexVisitor)
 }
+
+pub(crate) fn deserialize_jsonpath<'de, D>(deserializer: D) -> Result<JsonPathInst, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    deserializer.deserialize_str(JSONPathVisitor)
+}
+
+struct JSONPathVisitor;
+
+impl<'de> serde::de::Visitor<'de> for JSONPathVisitor {
+    type Value = JsonPathInst;
+
+    fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+        write!(formatter, "a string that begins with '$'")
```

Comment:
maybe a more explicit error message, like saying we expect a JSON path here?

Reply:
fixed. Thank you